src/classes/Math/Quaternion.js

```diff
@@ -100,6 +104,95 @@ Goblin.Quaternion.prototype = {
 		dest.z = iz * qw + iw * -qz + ix * -qy - iy * -qx;
 	},
 
+	setFromMat4: function (m) {
+        var m00, m01, m02, m10, m11, m12, m20, m21, m22,
+            tr, s, rs, lx, ly, lz;
+
+        // Cache matrix values for super-speed
+        m00 = m.e00;
+        m01 = m.e10;
+        m02 = m.e20;
+        m10 = m.e01;
+        m11 = m.e11;
+        m12 = m.e21;
+        m20 = m.e02;
+        m21 = m.e12;
+        m22 = m.e22;
+
+        // Remove the scale from the matrix
+        lx = 1 / Math.sqrt(m00 * m00 + m01 * m01 + m02 * m02);
```

Comment:
Can it be zero division?

Reply:
The code is actually courtesy of playcanvas :) Answering your question – yes, zero division can happen here in case the matrix is singular, but grabbing the rotation from such a matrix is not possible anyway.